workflow/controller/operator.go

```diff
@@ -1117,6 +1124,18 @@ func (woc *wfOperationCtx) podReconciliation(ctx context.Context) error {
 		node, err := woc.wf.Status.Nodes.Get(nodeID)
 		if err == nil {
 			if newState := woc.assessNodeStatus(pod, node); newState != nil {
+				if newState.Succeeded() {
+					tmpl, err := woc.GetNodeTemplate(node)
+					if err != nil {
+						return
```

Comment:
would this be an error that we should log?

Reply:
OK